builtin/logical/pki/cert_util.go

```diff
@@ -97,24 +133,30 @@ func fetchCertBySerial(req *logical.Request, prefix, serial string) (*logical.St
 // Given a set of requested names for a certificate, verifies that all of them
 // match the various toggles set in the role for controlling issuance.
 // If one does not pass, it is returned in the string argument.
-func validateCommonNames(req *logical.Request, commonNames []string, role *roleEntry) (string, error) {
+func validateNames(req *logical.Request, names []string, role *roleEntry) (string, error) {
 	hostnameRegex, err := regexp.Compile(`^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$`)
```

Comment:
The problem is that we can't use `\w` since that includes underscores. So it's just longhand. But it's simply saying "Any alphanumeric string, separated by dots, that can include hyphens but not as the first or last character of each string".